### PR TITLE
refactor: move timestamp util functions into class

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem

- timestamp handling is a mess
- make a class that handles the parameter determination
- no functionality is changed. All the current tests should pass

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by benjackwhite** [comment]
May be worth having a look a [this issue](https://github.com/PostHog/posthog/issues/11492) if you are working on anything related to dates. I tried to deep dive into why weird things happen with timezones. Couldn't find a solution but perhaps you will have better luck!

**Review by Twixes — COMMENTED** [review]
Looks reasonable! Only unsure about some naming.
This just reminded me of my major gripe with `TIME_IN_SECONDS`… it's forever assumed that every month is 30 days, which can result in some month-based Trends results to be off. Separate problem though.

File: posthog/queries/timestamp_query.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+from dateutil.relativedelta import relativedelta
+from django.utils import timezone
+from rest_framework.exceptions import ValidationError
+
+from posthog.models.team import Team
+from posthog.queries.util import PERIOD_TO_TRUNC_FUNC, TIME_IN_SECONDS, format_ch_timestamp, get_earliest_timestamp
+
+
+class TimestampQuery:
```
[the remark this answers is not in the document]
Comment (EDsCODE) [reply posthog/queries/timestamp_query.py:13]: yup thanks for calling this out i was being lazy :/

File: posthog/queries/timestamp_query.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    def num_intervals(self):
+
+        if self._filter.interval == "month":
+            rel_delta = relativedelta(self._end_time.replace(day=1), self._start_time.replace(day=1))
+            return (rel_delta.years * 12) + rel_delta.months + 1
+
+        return int(self.time_difference.total_seconds() / TIME_IN_SECONDS[self._filter.interval]) + 1
+
+    @cached_property
+    def should_round(self):
```
[the remark this answers is not in the document]
Comment (EDsCODE) [reply posthog/queries/timestamp_query.py:122]: it is slightly strange but the timezone_date_clause accounts for truncation too which is why it's "rounding"

**Comment by EDsCODE** [comment]
Will address https://github.com/PostHog/posthog/issues/11492 and TIME_IN_SECONDS issues separately. This PR is meant to just refactor without changing functionality!

**Pull request merged**